config/waagent-extn.logrotate

```diff
@@ -1,4 +1,4 @@
-/var/log/azure/*/*.log {
+/var/log/azure/*/CommandExecution.log {
```

Comment:
will this produce a conflict if an extension creates a rule for /var/log/azure/<extension>/*?

Reply:
Yes, but don't think there's a way around that short of moving the location of the `CommandExecution.log` file, assuming that we still want it `logrotate`d. I've verified that the original extensions we conflicted with do not have that behavior, and I don't think it's viable to test every possible extension. The best bet (in my opinion) is to fix the conflicting configuration we already have with OMS/LAD and keep an eye on the situation in case any new conflicts pop up.

Reply:
Monitoring conflicting extensions is not feasible. To address the issue raised by OMS/LAD we can simply remove the rule altogether.

Reply:
We've had issues raised for the `CommandExecution.log` file growing too large, so it seems like it would require logrotate. I'm not sure if I see any alternative to some form of this rule.

Reply:
If there is a conflict then we are preventing logs from being rotated so then end result may be the same (using too much disk).

Reply:
As far as the size issue, some things to consider:
* The way the code is written, anything logged in ExtHandlerInstance will be echoed to this file. That is not appropriate, the majority of these messages correspond to the operation of the agent, not of the extension. Cleaning this up would reduce the file size significantly.
* Some extensions don't seem to have this file. Not sure if we have a bug or if extensions are deleting it (I checked custom script and run command)
* I would not mind removing the log altogether, since everything here is already in the agent's log
* If publishers really want this file they could rotate it
* If they can't rotate it and we really want to keep this file we could add special code for this log to truncate it to a small size